Mathlib/CategoryTheory/Category/Twop.lean

```diff
@@ -0,0 +1,220 @@
+Authors: Yaël Dillies
+
+! This file was ported from Lean 3 source module category_theory.category.Twop
+! leanprover-community/mathlib commit c8ab806ef73c20cab1d87b5157e43a82c205f28e
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.CategoryTheory.Category.Bipointed
+import Mathlib.Data.TwoPointing
+
+/-!
+# The category of two-pointed types
+
+This defines `Twop`, the category of two-pointed types.
+
+## References
+
+* [nLab, *coalgebra of the real interval*]
+  (https://ncatlab.org/nlab/show/coalgebra+of+the+real+interval)
+-/
+
+
+open CategoryTheory Option
+
+universe u
+
+variable {α β : Type _}
+
+/-- The category of two-pointed types. -/
+structure Twop : Type (u + 1) where
+  X : Type u
+  toTwoPointing : TwoPointing X
+set_option linter.uppercaseLean3 false in
+#align Twop Twop
+
+namespace Twop
+
+instance : CoeSort Twop (Type _) :=
+  ⟨X⟩
+
+-- attribute [protected] Twop.X -- Porting note: protected attribute doesn't work
+
+/-- Turns a two-pointing into a two-pointed type. -/
+def of {X : Type _} (to_two_pointing : TwoPointing X) : Twop :=
+  ⟨X, to_two_pointing⟩
+set_option linter.uppercaseLean3 false in
+#align Twop.of Twop.of
+
+@[simp]
+theorem coe_of {X : Type _} (to_two_pointing : TwoPointing X) : ↥(of to_two_pointing) = X :=
+  rfl
+set_option linter.uppercaseLean3 false in
+#align Twop.coe_of Twop.coe_of
+
+alias of ← _root_.TwoPointing.Twop
+set_option linter.uppercaseLean3 false in
+#align two_pointing.Twop TwoPointing.Twop
+
+instance : Inhabited Twop :=
+  ⟨of TwoPointing.bool⟩
+
+/-- Turns a two-pointed type into a bipointed type, by forgetting that the pointed elements are
+distinct. -/
+noncomputable def toBipointed (X : Twop) : Bipointed :=
+  X.toTwoPointing.toProd.Bipointed
+set_option linter.uppercaseLean3 false in
+#align Twop.to_Bipointed Twop.toBipointed
+
+@[simp]
+theorem coe_toBipointed (X : Twop) : ↥X.toBipointed = ↥X :=
+  rfl
+set_option linter.uppercaseLean3 false in
+#align Twop.coe_to_Bipointed Twop.coe_toBipointed
+
+noncomputable instance largeCategory : LargeCategory Twop :=
+  InducedCategory.category toBipointed
+set_option linter.uppercaseLean3 false in
+#align Twop.large_category Twop.largeCategory
+
+noncomputable instance concreteCategory : ConcreteCategory Twop :=
+  InducedCategory.concreteCategory toBipointed
+set_option linter.uppercaseLean3 false in
+#align Twop.concrete_category Twop.concreteCategory
+
+noncomputable instance hasForgetToBipointed : HasForget₂ Twop Bipointed :=
+  InducedCategory.hasForget₂ toBipointed
+set_option linter.uppercaseLean3 false in
+#align Twop.has_forget_to_Bipointed Twop.hasForgetToBipointed
+
+
+/-- Swaps the pointed elements of a two-pointed type. `TwoPointing.swap` as a functor. -/
+@[simps]
+noncomputable def swap : Twop ⥤ Twop where
+  obj X := ⟨X, X.toTwoPointing.swap⟩
+  map f := ⟨f.toFun, f.map_snd, f.map_fst⟩
+set_option linter.uppercaseLean3 false in
```

Comment:
Perhaps you can turn off this linter more globally for the file, since it seems to be needed every time.